### PR TITLE
Projectile Trap

### PR DESCRIPTION
**Pull request by Pukami**
Major Changes
- Made base classes for Traps, which extends to Collision Traps and Projectile Traps. 
- Added an Arrow Trap that can shoot an arrow every designated time interval, or by using a raycast trigger.
- Added a sync position function that moves the object's transform to the rigid body's. 

Minor Changes
- Small fix to chest not opening sometimes
- Made sure vectors are normalized in transform
- Added a distance parameter to raycasts. 

Future Changes
- Make projectile traps have collider-based trigger regions.

### CODE REVIEW

File: src/game_objects/traps/arrow.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        // 5 Seconds time out
+        time_out_ms = 5000;
+    }
+}
+
+void Arrow::play_trigger_animation() {
+    // This is where a cool animation/effects would play once the arrow hits...
+
+    // IF WE HAD ONE
```
Comment (richard1) [inline src/game_objects/traps/arrow.cpp:34]: lol

**Review by richard1 — COMMENTED** [review]
Looks fine to me, nice job on the projectile pool.

Gonna let someone on graphics add their review too.

**Comment by jytang** [comment]
Could we get these PRs in quicker and smaller? i.e. should merge in when approved and stable, but in small modular chunks.

**Comment by Pukami** [comment]
It's not really a big PR. It's just that it's going on top of my Essence Update thing now that has been waiting for approval.

**Pull request merged**